CMakeLists.txt

```diff
@@ -932,6 +932,9 @@ endif()
 
 if(CURL_USE_RUSTLS)
   find_package(Rustls REQUIRED)
```

Comment:
I tried `find_package(Rustls REQUIRED 0.15)` first, but it didn't work in practice, I'm not sure why. Checking `RUSTLS_VERSION` manually seems to work, and let me write a nice message.

Reply:
That's a feature of the non-`pkg-config` detection branch,
via `find_package_handle_standard_args` in `CMake/FindRustls.cmake`.
But, that would also need for us to detect the version manually, e.g.
from a header file (there wasn't any there last time I checked).

`RUSTLS_VERSION` on the hand only works when detected via
`pkg-config` (defined by `pkg_check_modules`). In other cases
it may always fail now due to an empty value.

Reply:
Thanks for taking a look. I'm not surprised I missed some details :laughing: 

> But, that would also need for us to detect the version manually, e.g. from a header file (there wasn't any there last time I checked).

I could look at adding version `#define`'s into `rustls.h` for a future release. In the meantime could we adopt the same technique as the `m4` and look for a required function that's only present in 0.15? Open to your advice here.

Reply:
edit: if such version info isn't available in a header or other obvious
place, maybe doing a check for `rustls_supported_hpke` here as
well would solve it (though with the cost of `configure` step performance).

Reply:
@vszakats I pushed an update reworking the CMake version check. The pkg-config branch uses a `RUSTLS_VERSION` check like before, and the non-pkg-config branch uses `check_symbol_exists(rustls_supported_hpke "rustls.h" RUSTLS_HAS_HPKE)`

Does that look closer to what you had in mind?